### ISSUE TITLE
Result Reproducibility

### ISSUE THREAD

**Issue by phillipecardenuto** [facebookresearch/sscd-copy-detection#5]
Hi, @edpizzi congrats on your work!
I am trying to reproduce the results from your paper, and I checked that the code provided in this repo will have different results from the paper.

Would it be possible to provide the results of the exact training code from this repo on the DISC dataset?
If possible, could you provide these results using the same parameter described in the paper:
--entropy_weight=30 --augmentations=ADVANCED --mixup=true 

Kind regards,
João Phillipe

**Comment by edpizzi** [comment]
Thanks for your question and for your interest in our work @phillipecardenuto!

We expect some amount of deviation from paper results. We performed the paper experiments on an internal codebase, and this repository is a reimplementation. The only difference we know of are different augmentation files (eg. using freely available fonts in this version).

Looking at my notes from before we released this codebase, I have the following results recorded:

This OSS codebase:
micro AP: 62.3
micro AP SN: 72.9

The corresponding paper results:
micro AP: 61.5
micro AP SN: 72.5

Are you seeing modest deviations in accuracy, or more significant changes?

**Comment by phillipecardenuto** [comment]
Hi @edpizzi ,

Sorry for the late reply.

Due to my limited infrastructure, I had some trouble using the same batch size as used in the paper and in this repo. Thus, I could not reproduce the exact same experiments as you did.

In my case, the largest batch size I could use is 1200 (yours is 4096).
The results I got were similar to yours, and the difference might be explained by the different batch sizes.

My Results:
micro AP: 59.0
micro AP SN: 71.0

Thanks for your quick answer and for providing your results.

Kind Regards,
João Phillipe

**Comment by edpizzi** [comment]
Yea, that’s close, and we expect a bit of degradation at smaller batch sizes.

In the paper supplemental we explore batch size and see ~2% degradation by batch size 2048, but we explored it at entropy weight 15.

We could consider changing this to support larger batch sizes using gradient checkpointing. Marking as closed for now.
